### PR TITLE
Add custom hooks user API

### PR DESCRIPTION
**Pull request by oetr**

### CODE REVIEW

File: examples/custom-hooks/custom-hooks.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * The original function is never called.
+ * The hook does nothing other than logging.
+ * This can be useful for bypassing the fuzzing blockers to achieve
+ * coverage of more interesting functions.
+ */
+registerReplaceHook(
+	"JpegImage.jpegImage.buildHuffmanTable",
+	"jpeg-js",
+	false,
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (bertschneider) [inline examples/custom-hooks/custom-hooks.js:64]: Disabling the warning is not required here, as there are no args, but in the first hook.

File: packages/hooking/manager.ts [hunk 2]
```diff
@@ -181,3 +181,13 @@ export class HookManager {
 }
 
 export const hookManager = new HookManager();
+
+// convenience function to register hooks
+export function registerReplaceHook(
```
Comment (bertschneider) [inline packages/hooking/manager.ts:186]: Should we add convenience functions for `Before` and `After` as well?

File: docs/fuzz-settings.md [hunk 3]
```diff
@@ -32,3 +32,56 @@ Base64: 4+Pj4+Pj4+Pj4wo=
...
+
+```javascript
+const { registerReplaceHook } = require("@jazzer.js/hooking");
+```
+
+To register a custom hook, the function `registerReplaceHook` can be used:
+
+```typescript
+function registerReplaceHook(
+	target: string, // target function name that we want to hook
```
Comment (bertschneider) [inline docs/fuzz-settings.md:77]: We should also explain that the target name for nested "structure" is constructed by concatenating the individual names using dots, like `SomeClass.someMethod.someInnerFunction`.

File: packages/hooking/manager.ts (hunk 2 above)
Comment (oetr) [reply packages/hooking/manager.ts:195]: Added!

File: packages/core/core.ts [hunk 4]
```diff
@@ -39,15 +40,18 @@ function initFuzzing(options: Options): fuzzer.FuzzFn {
 	globalThis.Fuzzer = fuzzer.fuzzer;
 	//TODO make sure that all sanitizers are registered at this point
 	globalThis.HookManager = hooking.hookManager;
+	// load each custom hook file
+	options.customHooks.forEach((customHook) => {
+		const fuzzFn = importModule(customHook);
```
Comment (bertschneider) [inline packages/core/core.ts:45]: No need to assign the module to a local variable.

**Review by bertschneider — APPROVED** [review]
Great, thank you!

File: packages/core/core.ts (hunk 4 above)
Comment (oetr) [reply packages/core/core.ts]: Updated and moved into 17c51ef.

**Pull request merged**